state_machine_design.py

```diff
@@ -118,30 +118,45 @@ def process(self, sm):
         print("CHECK_TIME process func")
         print("ens index: " + str(sm.ens_index))
 
+        # this is a small window where if the scheduler wakes up
+        # slightly early from sleep we allow it to run the ensemble anyway
+        TIME_BUFFER = 5
+
+
         if sm.ens_index < len(sm.ens_list):
             # read time from ensemble and compare to current_time
             now = time.localtime()
-            curr_time_seconds = hmsToSeconds(now.tm_hour, now.tm_min, now.tm_sec)
+            curr_time_seconds = hms_to_seconds(now.tm_hour, now.tm_min, now.tm_sec)
 
             nearest_ens_time = sm.ens_list[sm.ens_index]["start_time"]
 
+                
             if nearest_ens_time < curr_time_seconds:
-                print("time past current ens, skipping")
-                self.check_time_ctrl = SKIP
-                sm.err_code = MISSED_ENS
-                # sm.err_msg = "Current ensemble past time, skipping ensemble "
-                # + sm.ens_list[sm.ens_index]["title"] + " at " + curr_time_seconds
-            elif nearest_ens_time == curr_time_seconds:
+                print("time past current ens, checking if should skip")
+                if sm.rst == True:
+                    if now.tm_wday == sm.day_of_ens:
+                        self.check_time_ctrl = WAIT
+                    else:
+                        self.check_time = SKIP
+                        sm.err_code = MISSED_ENS
+                else:
+                    self.check_time_ctrl = SKIP
+                    sm.err_code = MISSED_ENS
```

Comment:
Is there a difference between `self.check_time` and `self.check_time_ctrl`? I don't see the former anywhere else.
If they're the same, can we combine this into a single if-else statement, with the two conditions ANDed together, since the elses are the same?

Reply:
Oop, apologies. I was writing late last night and typo'd that. Yeah I can fix that and combine them as well.